src/misc/cube-env-map.js

```diff
@@ -0,0 +1,133 @@
+/**
```

Comment:
This module is re-instated as it was previously, except for a couple of small bug fixes commented inline